pkg/database/extensions.go

```diff
@@ -0,0 +1,124 @@
+package database
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"encoding/base64"
+
+	"github.com/sirupsen/logrus"
+	"github.com/ugorji/go/codec"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/encrypt"
+)
+
+var _ codec.InterfaceExt = (*SecureByteExt)(nil)
+
+type SecureByteExt struct {
+	Log    *logrus.Entry
+	Cipher encrypt.Cipher
+}
+
+func (s SecureByteExt) ConvertExt(v interface{}) interface{} {
+	data := v.(api.SecureByte)
+	var encrypted []byte
+	if s.Cipher != nil {
+		var err error
+		encrypted, err = s.Cipher.Encrypt(data)
+		if err != nil {
+			s.Log.Error(err)
+		}
+	}
+
+	return encrypted
+}
+func (s SecureByteExt) UpdateExt(dest interface{}, v interface{}) {
+	output := dest.(*api.SecureByte)
+	var decrypted []byte
+	if s.Cipher != nil {
+		var err error
+		decrypted, err = s.Cipher.Decrypt(v.([]byte))
+		if err != nil {
+			s.Log.Error(err)
+		}
+	}
+
+	// TODO(MJ): Remove this when galaxy is in a stable period
+	// This is for backwards compatibility. If we detected base64 encoded
+	// string from NoopExt after decrypt, this means its an old record and
+	// we need to decode.
+	if IsBase64(string(decrypted)) {
+		decoded, err := base64.StdEncoding.DecodeString(string(decrypted))
+		if err != nil {
+			s.Log.Error(err)
+		}
+		*output = []byte(decoded)
+		return
+	}
+	*output = decrypted
+}
+
+var _ codec.InterfaceExt = (*SecureStringExt)(nil)
+
+type SecureStringExt struct {
+	Log    *logrus.Entry
+	Cipher encrypt.Cipher
+}
+
+func (s SecureStringExt) ConvertExt(v interface{}) interface{} {
+	data := v.(api.SecureString)
+	var encrypted []byte
+	if s.Cipher != nil {
+		var err error
+		encrypted, err = s.Cipher.Encrypt([]byte(data))
+		if err != nil {
+			s.Log.Error(err)
+		}
+	}
+
+	return encrypted
+}
+func (s SecureStringExt) UpdateExt(dest interface{}, v interface{}) {
+	output := dest.(*api.SecureString)
+	var decrypted []byte
+	if s.Cipher != nil {
+		var err error
+		decrypted, err = s.Cipher.Decrypt(v.([]uint8))
+		if err != nil {
+			s.Log.Error(err)
+		}
+		// TODO(MJ): Remove this when galaxy is in a stable period
+		// This is for backwards compatibility. If we detected base64 encoded
+		// string from NoopExt after decrypt, this means its an old record and
+		// we need to decode.
+		if IsBase64(string(decrypted)) {
+			decoded, err := base64.StdEncoding.DecodeString(string(decrypted))
+			if err != nil {
+				s.Log.Error(err)
+			}
+			*output = api.SecureString(decoded)
+			return
+		}
```

Comment:
what's the case when this would happen?

Reply:
removed in favor of logic in `rawByteExt`